core/include/thread.h

```diff
@@ -195,61 +289,21 @@ struct _thread {
 #define THREAD_AUTO_FREE                (2)
 
 /**
- * @brief Do not automatically call thread_yield() after creation
+ * @brief Do not automatically call thread_yield() after creation: the newly
+ *        created thread will not run immediately
  */
 #define THREAD_CREATE_WOUT_YIELD        (4)
 
  /**
   * @brief Write markers into the thread's stack to measure stack usage (for
-  *        debugging)
+  *        debugging and profiling purposes)
   */
 #define THREAD_CREATE_STACKTEST         (8)
 /** @} */
 
 /**
  * @brief Creates a new thread
  *
- * Creating a new thread is done in two steps:
- * 1. the new thread's stack is initialized depending on the platform
- * 2. the new thread is added to the scheduler and the scheduler is run (if not
- *    indicated otherwise)
- *
- * As RIOT is using a fixed priority scheduling algorithm, threads are
- * scheduled based on their priority. The priority is fixed for every thread
- * and specified during the threads creation by the *priority* parameter.
- *
- * A low value for *priority* number means the thread having a high priority
- * with 0 being the highest possible priority.
- *
- * The lowest possible priority is *THREAD_PRIORITY_IDLE - 1*. The value is depending
- * on the platforms architecture, e.g. 30 in 32-bit systems, 14 in 16-bit systems.
- *
- * @note Assigning the same priority to two or more threads is usually not a
- *       good idea. A thread in RIOT may run until it yields (@ref
- *       thread_yield) or another thread with higher priority is runnable (@ref
- *       STATUS_ON_RUNQUEUE) again. Having multiple threads with the same
- *       priority may make it difficult to determine when which of them gets
- *       scheduled and how much CPU time they will get. In most applications,
- *       the number of threads in application is significantly smaller than the
- *       number of available priorities, so assigning distinct priorities per
- *       thread should not be a problem. Only assign the same priority to
- *       multiple threads if you know what you are doing!
- *
- *
- * In addition to the priority, the *flags* argument can be used to alter the
- * newly created threads behavior after creation. The following flags are available:
- *  - THREAD_CREATE_SLEEPING    the newly created thread will be put to sleeping
- *                              state and must be woken up manually
- *  - THREAD_CREATE_WOUT_YIELD  the newly created thread will not run
- *                              immediately after creation
- *  - THREAD_CREATE_STACKTEST   write markers into the thread's stack to measure
- *                              the stack's memory usage (for debugging and
- *                              profiling purposes)
- *
- * @note Currently we support creating threads from within an ISR, however it
- *       is considered to be a bad programming practice and we strongly discourage
- *       it.
- *
```

Comment:
The reason I moved this section upward was that it seemed a little odd to me to have in-depth explanations of the scheduling/flags/thread creation in a function documentation. I'd argue that somebody who'd rather read the thread_create() doc will (for example) look up how the flags are defined by themselves. However, I can see that the ISR note is important for topic oriented readers also, so if that's what you had in mind I'll be happy to replicate it in the thread_create() comment as well :)